sylt-compiler/src/typechecker.rs

```diff
@@ -178,12 +177,45 @@ impl<'c> TypeChecker<'c> {
                 let ret = Box::new(self.solve_generics_recursively(span, generics, a_ret, b_ret)?);
                 Type::Function(args, ret)
             }
+            (a, Type::Union(b)) => {
+                // This is technically wrong, since we might do guesses here that aren't valid.
+                // But it works for our uses.
```

Comment:
Maybe an example where this is wrong? It might help if something went funky-wunky here.